api/v1alpha1/argocd_types.go

```diff
@@ -341,6 +341,9 @@ type ArgoCDRepoSpec struct {
 	// MountSAToken describes whether you would like to have the Repo server mount the service account token
 	MountSAToken bool `json:"mountsatoken,omitempty"`
 
+	// Replicas defines the number of replicas for argocd-repo-server
+	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
Any reasons why this is a pointer?

Reply:
because the typed value could also be nil. Should it not be a pointer?

Reply:
What is the effective number of replica when the value of this int32 pionter is nil?

Reply:
Oh I see what you mean actually, previously I thought this field was optional but I realize now that even if you get rid of it it will still come back. 

However, we have a line that says `deploy.Spec.Replicas = getArgoCDRepoServerReplicas(cr)` where `deploy.Spec.Replicas` is of type `*int32`, but `getArgoCDRepoServerReplicas` would return a value of type `int32` if we removed the pointer. (same for argocd-server)

Reply:
Ok.  thanks.